### ISSUE TITLE
unarchive module add directory_mode parameter

### ISSUE THREAD

**Issue by edysli** [ansible/ansible#19760]
_From @edysli on February 16, 2016 8:47_

##### Issue Type:

Feature Idea
##### Component Name:

unarchive
##### Ansible Version:

```
ansible 2.0.0.2
  config file = /etc/ansible/ansible.cfg
  configured module search path = Default w/o overrides
```
##### Summary:

Currently, using unarchive with `mode=0644` creates directories without the execute permission, making them impossible to `cd` into. It would be nice to have a `directory_mode` parameter for unarchive, just like the copy module has, in order to set the proper mode on directories (`directory_mode=0755`).

_Copied from original issue: ansible/ansible-modules-core#3048_

**Comment by edysli** [comment]
_From @ansibot on July 31, 2016 18:4_



@dagwieers, @pileofrogs, ping. This issue is waiting on your response.
[click here for bot help](https://github.com/jctanner/ansibullbot/blob/ISSUE_TRIAGE_CLASS/ISSUE_HELP.md)

**Comment by edysli** [comment]
_From @dagwieers on August 8, 2016 22:53_

I am all for this !

**Comment by edysli** [comment]
_From @ansibot on August 8, 2016 22:59_

@edysli we are unable to validate that "" is an existing ansible module.

We are expecting a component name that matches something that can be found in:
- http://docs.ansible.com/ansible/list_of_all_modules.html

Please revise the component name in the description.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by edysli** [comment]
@ansibot component name "unarchive" added to description

**Comment by edysli** [comment]
_From @dagwieers on August 19, 2016 11:26_

needs_contributor

**Comment by edysli** [comment]
_From @ansibot on December 9, 2016 19:52_

This repository has been locked. All new issues and pull requests should be filed in https://github.com/ansible/ansible

Please read through the <a href="http://docs.ansible.com/ansible/dev_guide/repomerge.html#move-issues-and-prs-to-new-repo">repomerge</a> page in the dev guide. The guide contains links to tools which automatically move your issue or pull request to the ansible/ansible repo.

**Comment by sijis** [comment]
I'm using a fairly recent version of ansible (2.4.2.0 with python 2.7.5) and I'm not longer seeing that behavior.

**playbook**
```
---
- hosts: localhost
  vars:
    issue: 19760
  tasks:
    - name: Create directory
      file:
        state: directory
        path: "{{item}}"
      with_items:
        - "/tmp/test-{{issue}}"
        - "files/test-{{issue}}"
    - name: Create zip
      shell: "touch files/test-{{issue}}/{0..10} && zip -qq -r files/test-{{issue}}.zip files/test-{{issue}}"
    - name: Unpack zip file
      unarchive:
        src: "files/test-{{issue}}.zip"
        dest: "/tmp/test-{{issue}}"
        copy: no
        #extra_opts:
        #  - "-qq"
        mode: 0644
      register: testing
```

**result**
```
[root@0df32e44ec62 /]# ls -l /tmp/test-19760/ -d
drwxr-xr-x. 3 root root 4096 May 18 04:35 /tmp/test-19760/
[root@0df32e44ec62 /]# ls -l /tmp/test-19760/files/ -d
drwxr-xr-x. 3 root root 4096 May 18 04:35 /tmp/test-19760/files/
[root@0df32e44ec62 /]# ls -l /tmp/test-19760/files/test-19760/ -d
drw-r--r--. 2 root root 4096 May 18 04:35 /tmp/test-19760/files/test-19760/
[root@0df32e44ec62 /]# ls -l /tmp/test-19760/files/test-19760/ | head
total 0
-rw-r--r--. 1 root root 0 May 18 04:35 0
-rw-r--r--. 1 root root 0 May 18 04:35 1
-rw-r--r--. 1 root root 0 May 18 04:35 10
-rw-r--r--. 1 root root 0 May 18 04:35 2
-rw-r--r--. 1 root root 0 May 18 04:35 3
-rw-r--r--. 1 root root 0 May 18 04:35 4
-rw-r--r--. 1 root root 0 May 18 04:35 5
-rw-r--r--. 1 root root 0 May 18 04:35 6
-rw-r--r--. 1 root root 0 May 18 04:35 7
```
I do realize this was initially brought up a few years ago, but it does appear it was addressed between 2.0 and 2.2 ansible releases when installing those versions via easy_install/pip.

**Comment by peruzzof** [comment]
With a .tgz file it will apply the permission listed in "mode" to directory and files. So we must grant an useless (and sometimes dangerous) execution permission to regular files.

**Comment by ansibot** [comment]
Thank you very much for your submission to Ansible. It means a lot to us that you've taken time to contribute.

Unfortunately, this issue has been open for some time while waiting for a contributor to take it up but there does not seem to have been anyone that did so. So we are going to close this issue to clear up the queues and make it easier for contributors to browse possible implementation targets.

However, we're absolutely always up for discussion. Because this project is very active, we're unlikely to see comments made on closed tickets and we lock them after some time. If you or anyone else has any further questions, please let us know by using any of the communication methods listed in the page below:

   * https://docs.ansible.com/ansible/latest/community/communication.html

In the future, sometimes starting a discussion on the development list prior to proposing or implementing a feature can make getting things included a little easier, but it's not always necessary.

Thank you once again for this and your interest in Ansible!

[click here for bot help](https://github.com/ansible/ansibullbot/blob/devel/ISSUE_HELP.md)
